spring-integration-smb/src/main/java/org/springframework/integration/smb/session/SmbSession.java

```diff
@@ -458,7 +454,7 @@ private SmbFile createSmbFileObject(String path, Boolean isDirectory) throws IOE
 			return this.smbShare;
 		}
 
-		SmbFile smbFile = new SmbFile(this.smbShare, cleanedPath, SmbFile.FILE_SHARE_READ);
+		SmbFile smbFile = new SmbFile(this.smbShare, cleanedPath);
```

Comment:
Why have you dropped that `SmbFile.FILE_SHARE_READ` option?
Isn't there some alternative any more?

Reply:
Any insights here, please?

Reply:
The constructor for SmbFile( SmbFile context, String name, int shareAccess ) has been removed from the latest version of jCIFS, and the closest one now available is public SmbFile ( SmbResource context, String name ).  SmbFile now implements SmbResource. SmbResource is also new.

Reply:
That's OK with me, but how about that `shareAccess`? Isn't there a way to configure it any more?
How can we be sure that an appropriate access is granted?

Reply:
From what I can tell by reading through the jCIFS code that the share attribute was used in a constructor to the old SmbComNTCreateAndX class that is now in a separate internal\smb1 package off the root jcifs package. I think that is why this old constructor was removed. It might be redundant now. I'm still looking to see if there is a comparable way to set the share to "file share read".

Reply:
Ok pouring through the latest code for jCIFS appears to show that this was an old concept for SMB1, hence the reason it was removed. The new constructor without the SmbFile.FILE_SHARE_READ flag initializes a SMB2 file with flags SmbConstants.FILE_SHARE_READ | SmbConstants.FILE_SHARE_WRITE. It looks like this old feature from SMB1 has been disabled.

The old SmbFile has been moved into a new jcifs.smb1.smb1 package along with all the old code for SMB1. Since we are upgrading this extension to use the latest SMB protocols then I think it is no longer necessary to set the share access permissions. The latest version doesn't appear to allow this. Also all the JUnit tests in jCIFS do not set this share flag anymore, nor is there an example I can use to set this flag for SmbConstants.FILE_SHARE_READ only.

Reply:
OK. Fair enough.

I really can just live with whatever is there by default.